### PR TITLE
Support passing props to tab routes

### PR DESCRIPTION
**Pull request by benstepp**
Basically when a JUMP occurs, a copy of the wrapped scene is pulled out of state.scenes and placed in the state tree with the props copied onto it. When no props are passed in the action, all props are removed as every jump grabs a new copy of the scene. 

I'm not too sure about the implementation, so any other ideas may be worth exploring.

```jsx
<Scene key="root" tabs>
  <Scene key="home" component={Home} />
  {/*...*/}
</Scene>

Actions.home({customProp: true})
// customProp = true
// DefaultRenderer renders Home with customProp: true

Actions.home()
// customProp = undefined
// DefaultRenderer renders Home without customProp key in props
```

Reducer.test.js has the tests relevant to this change. I was just familiarizing myself with the codebase in Actions.test.js.

### CODE REVIEW

File: src/Reducer.js [hunk 1]
```diff
@@ -163,7 +163,18 @@ function inject(state, action, props, scenes) {
...
+
+          // Pass the props to the wrapped tab scene
+          if (props.children) delete props.children;
+          const key = child.children[0].sceneKey;
+          const wrappedScene = scenes[key];
+          child.children = [{ ...props, ...wrappedScene }];
+        }
+      });
+
       assert(ind !== -1, `Cannot find route with key=${action.key} for parent=${state.key}`);
```
Comment (aksonov) [inline src/Reducer.js]: You should not modify existing state, but produce new one. Here i see you modify existing child within existing state var.

Comment (benstepp) [inline src/Reducer.js]: Good looking out. It just seemed like the way to go with all of the shallow slices pointing back to previous state objects, and direct setting of state vars [here](https://github.com/aksonov/react-native-router-flux/blob/master/src/Reducer.js#L178) and [here](https://github.com/aksonov/react-native-router-flux/blob/master/src/Reducer.js#L191).

This does give me a change to better cover edge cases though. I noticed when nesting tabs things stop behaving as expected. I'm thinking at least tests for double nested tab routes should cover it. I'll take another stab at it this weekend.

**Comment by nicholeuf** [comment]
This feature would be very nice to have!

**Comment by Herlevsen** [comment]
This is not working anymore
